contracts/token/ERC20/extensions/ERC20Permit.sol

```diff
@@ -70,8 +66,8 @@ abstract contract ERC20Permit is ERC20, IERC20Permit, EIP712 {
     /**
      * @dev See {IERC20Permit-nonces}.
      */
-    function nonces(address owner) public view virtual override returns (uint256) {
-        return _nonces[owner].current();
+    function nonces(address owner) public view virtual override(IERC20Permit, Nonces) returns (uint256) {
+        return super.nonces(owner);
     }
```

Comment:
Does it compile if we remove this function definition? Please try with the latest Solidity version. This is a requirement that was relaxed at some point.

Reply:
it does compile, but doesn't work properly always returning zero, tried with the latest version.

Reply:
That doesn't make sense to me... There is something wrong there.

Reply:
My bad, it doesn't compile